File: ai-service/routers/burnout.py
"""Burnout router mounted at /calculate-burnout in main.py."""

from fastapi import APIRouter

from schemas.burnout import BurnoutInput, BurnoutOutput
from services.burnout_calculator import BurnoutCalculatorService

router = APIRouter(tags=["Burnout"])

_calculator = BurnoutCalculatorService()


@router.post(
    "",
    response_model=BurnoutOutput,
    summary="Calculate burnout risk score",
    description=(
        "Accepts shift metrics and returns a burnout risk score (0-100), "
        "risk level (LOW/MEDIUM/HIGH/CRITICAL), a per-factor breakdown, "
        "and actionable recommendations."
    ),
)
async def calculate_burnout(payload: BurnoutInput) -> BurnoutOutput:
    """POST /calculate-burnout — compute burnout risk from shift data."""
    return _calculator.calculate(payload)

File: frontend/src/api/burnout.js
import api from './axios';

export const getBurnoutScores = async (params) => {
  const response = await api.get('/burnout/', { params });
  return response.data;
};

export const calculateBurnout = async (data) => {
  const response = await api.post('/burnout/calculate/', data || {});
  return response.data;
};
